cranelift/codegen/src/isa/x64/inst/args.rs

```diff
@@ -994,3 +995,36 @@ impl OperandSize {
         }
     }
 }
+
+/// An x64 memory fence kind.
+#[derive(Clone)]
+pub enum FenceKind {
+    MFence,
+    LFence,
+    SFence,
+}
+
+/// Atomic memory update operations.  FIXME: dup: identical to the aarch64 version
```

Comment:
Maybe hoist the aarch64 version then, and hoist it in the `machinst` code or a new `isa/common` directory?